plugins/inputs/uwsgi/uwsgi.go

```diff
@@ -0,0 +1,227 @@
+// Package uwsgi implements a telegraf plugin for
+// collecting uwsgi stats from the uwsgi stats server.
+package uwsgi
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+)
+
+var timeout = 5 * time.Second
+
+// uWSGI Server struct
+type Uwsgi struct {
+	Servers []string `toml:"server"`
+}
+
+// Errors is a list of errors accumulated during an interval.
+type Errors []error
+
+func (errs Errors) Error() string {
+	s := ""
+	for _, err := range errs {
+		if s == "" {
+			s = err.Error()
+		} else {
+			s = s + ". " + err.Error()
+		}
+	}
+	return s
+}
+
+// NestedError wraps an error returned from deeper in the code.
+type NestedError struct {
+	// Err is the error from where the NestedError was constructed.
+	Err error
+	// NestedError is the error that was passed back from the called function.
+	NestedErr error
+}
+
+// Error returns a concatenated string of all the nested errors.
+func (ne NestedError) Error() string {
+	return ne.Err.Error() + ": " + ne.NestedErr.Error()
+}
+
+// Errorf is a convenience function for constructing a NestedError.
+func Errorf(err error, msg string, format ...interface{}) error {
+	return NestedError{
+		NestedErr: err,
+		Err:       fmt.Errorf(msg, format...),
+	}
+}
+
+// Description returns the plugin description
+func (u *Uwsgi) Description() string {
+	return "Read uWSGI metrics."
+}
+
+// SampleConfig returns the sample configuration
+func (u *Uwsgi) SampleConfig() string {
+	return `
+    ## List with urls of uWSGI Stats servers. Url must match pattern:
+    ## scheme://address[:port]
+    ##
+    ## For example:
+    ## servers = ["tcp://localhost:5050", "http://localhost:1717", "unix:///tmp/statsock"]
+    servers = []
+`
+}
+
+// Gather collect data from uWSGI Server
+func (u *Uwsgi) Gather(acc telegraf.Accumulator) error {
+	var errs Errors
+	for _, s := range u.Servers {
+		n, err := url.Parse(s)
+		if err != nil {
+			return fmt.Errorf("Could not parse uWSGI Stats Server url '%s': %s", s, err)
+		}
+		if err := u.gatherServer(acc, n); err != nil {
+			errs = append(errs, Errorf(err, "server %s", s))
+		}
+
+	}
+	return errs
+}
+
+func (u *Uwsgi) gatherServer(acc telegraf.Accumulator, url *url.URL) error {
+	var err error
+	var r io.ReadCloser
+
+	switch url.Scheme {
+	case "unix":
+		r, err = net.DialTimeout(url.Scheme, url.Path, timeout)
+	case "tcp":
+		r, err = net.DialTimeout(url.Scheme, url.Host, timeout)
+	case "http":
+		resp, err := http.Get(url.String())
```

Comment:
I second that, with the `DefaultClient`, the timeout is set to 0, so no timeout. The gorouting in which this would run could be blocked for a significant time. 

Better  to use
```go
var httpClient = &http.Client{
  Timeout: time.Second * 30, // or some parameter as @gam-phon suggested
}
response,  err := httpClient.Get(url.String())
```